### PR TITLE
Add GEOS as an explicit runtime dependency

### PR DESCRIPTION
**Pull request by weiji14**
Explicitly pinning to GEOS 3.10 instead of relying on installing GEOS through GDAL only. This is effectively a re-render to build GMT on GEOS 3.10.0 instead of GEOS 3.9.1. Should make GMT play better with other geospatial packages like cartopy/cuspatial/etc.

Attempt at addressing https://github.com/GenericMappingTools/pygmt/issues/1600.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by weiji14** [comment]
@conda-forge-admin, please rerender

**Comment by weiji14** [comment]
Cc @meghanrjones, the macOS build worked on GEOS 3.10, and running the test commands `gmt coast` et al. seemed to work.

I noticed a lot of `is not an ELF` lines in the logs though at https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=401862&view=logs&j=58ac6aab-c4bc-5de2-2894-98e408cc8ec9&t=933f325c-924e-533d-4d95-e93b5843ce8b, could you examine the logs and see if those are normal and can be safely ignored?

**Review by seisman — APPROVED** [review]

File: recipe/meta.yaml
Comment (seisman) [inline recipe/meta.yaml:49]: Two suggestions:

1. Move `geos` up to lin 38, so that libraries `gdal` and `geos` are together
2. Also add `geos` as a `host` dependency at line24.

**Review by seisman — APPROVED** [review]
Let me know if you want me to merge this PR after all tests pass. Or we can wait for @meghanrjones's comments on the `is not an ELF` errors.

**Comment by weiji14** [comment]
> Let me know if you want me to merge this PR after all tests pass. Or we can wait for @meghanrjones's comments on the `is not an ELF` errors.

Let's just merge first, I don't think those are serious.

We should probably add Meghan as a feedstock maintainer here too (if she's ok ) to increase the bus factor.

**Review by leouieda — APPROVED** [review]
Good call on this 👍🏽

**Comment by meghanrjones** [comment]
Thanks @weiji14! I do not know how to interpret the `is not an ELF` statements since there is no library referenced in the warning.

**Pull request merged**
